pywemo/subscribe.py

```diff
@@ -35,32 +73,124 @@ def get_ip_address(host='1.2.3.4'):
+      trigger an event notification. This is implemented by configuring the
+      Wemo device to "control the state" of a virtual Wemo device. The virtual
+      device is implemented by this class.
+
+      The do_GET/do_POST/do_SUBSCRIBE methods below implement a virtual Wemo
+      device. The virtual device receives requests to change its state from
+      other Wemo devices on the network. When a Wemo device is configured to
+      change the state of the virutal device via a long press rule the following
+      sequence occurs:
+
+      1. The Wemo device will attempt to locate the virtual device on the
+      network. This is handled by the pywemo.ssdp.DiscoveryResponder class. See
+      the documentation there for more information about this step.
+
+      2. The Wemo device will fetch /setup.xml from do_GET to learn of the
+      virtual device details.
+
+      3. The Wemo device will subscribe to BinaryState notifications from the
+      virtual device. The virtual device does not send any BinaryState
+      notifications, but this step seems to be necessary before the next step
+      can happen. This step is implemented by the do_SUBSCRIBE method.
+
+      4. When a person presses the button on the Wemo for 2 seconds a long press
+      rule is triggered. If the long press rule is configured with an action
+      for the virtual device, the Wemo device will then call the do_POST method
+      to update the BinaryState of the virtual device. This doesn't actually
+      update any state, rather the virtual device then delivers the event
+      notification to any event listeners configured to receive events from the
+      pywemo SubscriptionRegistry. The event type for a long press action is
+      EVENT_TYPE_LONG_PRESS.
+    """
+
+    def do_NOTIFY(self):  # pylint: disable=invalid-name
         """Handle subscription responses received from devices."""
         sender_ip, _ = self.client_address
         outer = self.server.outer
         device = outer.devices.get(sender_ip)
-        content_len = int(self.headers.get('content-length', 0))
-        data = self.rfile.read(content_len)
         if device is None:
             LOG.warning('Received event for unregistered device %s', sender_ip)
         else:
-            # trim garbage from end, if any
-            data = data.decode("UTF-8").split("\n\n")[0]
-            doc = cElementTree.fromstring(data)
+            doc = self._get_xml_from_http_body()
             for propnode in doc.findall('./{0}property'.format(NS)):
                 for property_ in list(propnode):
                     text = property_.text
                     outer.event(device, property_.tag, text)
 
-        self.send_response(200)
-        self.send_header('Content-Type', 'text/html')
-        self.send_header('Content-Length', len(SUCCESS))
+        self._send_response(200, RESPONSE_SUCCESS)
+
+    def do_GET(self):  # pylint: disable=invalid-name
```

Comment:
I'd probably just go ahead and add all of these "do_UPPERCASE" names to the pylintrc file ignore list instead of locally disabling them here.

Reply:
See updated (second) comment on the previous line